ta4j-core/src/main/java/org/ta4j/core/BaseTradingRecord.java

```diff
@@ -136,6 +136,18 @@ public final boolean exit(int index, Decimal price, Decimal amount) {
         return trades;
     }
     
+    @Override
+    public List<Trade> getTrades(int beginIndex, int endIndex) {
+        List<Trade> trades = new ArrayList<Trade>();
+        for (Trade trade : this.trades) {
+            if (trade.getEntry().getIndex() < beginIndex || trade.getExit().getIndex() > endIndex) {
```

Comment:
null check?

Reply:
no null check needed. If user wants to get trades between null and/or null he will get is npe